pkg/controller/nginxingress/nginx_ingress_controller_test.go

```diff
@@ -1099,12 +1111,61 @@ func TestToNginxIngressConfig(t *testing.T) {
 				},
 			},
 			want: manifests.NginxIngressConfig{
-				ControllerClass: "approuting.kubernetes.azure.com/nicName",
-				ResourceName:    "controllerNamePrefix-0",
-				ServiceConfig:   &manifests.ServiceConfig{},
-				IcName:          "ingressClassName",
-				MaxReplicas:     20,
-				MinReplicas:     5,
+				ControllerClass:                "approuting.kubernetes.azure.com/nicName",
+				ResourceName:                   "controllerNamePrefix-0",
+				ServiceConfig:                  &manifests.ServiceConfig{},
+				IcName:                         "ingressClassName",
+				MaxReplicas:                    20,
+				MinReplicas:                    5,
+				TargetCPUUtilizationPercentage: defaultTargetCPUUtilization,
+			},
+		},
+		{
+			name: "custom fields with rapid threshold",
+			nic: &approutingv1alpha1.NginxIngressController{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "nicName",
+				},
+				Spec: approutingv1alpha1.NginxIngressControllerSpec{
+					IngressClassName:     "ingressClassName",
+					ControllerNamePrefix: "controllerNamePrefix",
+					Scaling: &approutingv1alpha1.Scaling{
+						Threshold: util.ToPtr(approutingv1alpha1.RapidThreshold),
+					},
+				},
+			},
+			want: manifests.NginxIngressConfig{
+				ControllerClass:                "approuting.kubernetes.azure.com/nicName",
+				ResourceName:                   "controllerNamePrefix-0",
+				ServiceConfig:                  &manifests.ServiceConfig{},
+				IcName:                         "ingressClassName",
+				MaxReplicas:                    defaultMaxReplicas,
+				MinReplicas:                    defaultMinReplicas,
+				TargetCPUUtilizationPercentage: rapidTargetCpuUtilization,
+			},
+		},
+		{
+			name: "custom fields with steady threshold",
+			nic: &approutingv1alpha1.NginxIngressController{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "nicName",
+				},
+				Spec: approutingv1alpha1.NginxIngressControllerSpec{
+					IngressClassName:     "ingressClassName",
+					ControllerNamePrefix: "controllerNamePrefix",
+					Scaling: &approutingv1alpha1.Scaling{
+						Threshold: util.ToPtr(approutingv1alpha1.SteadyThreshold),
+					},
+				},
+			},
+			want: manifests.NginxIngressConfig{
+				ControllerClass:                "approuting.kubernetes.azure.com/nicName",
+				ResourceName:                   "controllerNamePrefix-0",
+				ServiceConfig:                  &manifests.ServiceConfig{},
+				IcName:                         "ingressClassName",
+				MaxReplicas:                    defaultMaxReplicas,
+				MinReplicas:                    defaultMinReplicas,
+				TargetCPUUtilizationPercentage: steadyTargetCPUUtilization,
 			},
```

Comment:
Can you add a custom fields with balanced threshold case as well

Reply:
done